iptables-module/xt_mpeg2ts.c

```diff
@@ -271,8 +273,9 @@ struct pid_data_t {
 
 /* Data to match a stream / connection */
 struct mpeg2ts_stream_match { /* Like xt_hashlimit: dsthash_dst */
-	__be32 dst_addr; /* MC addr first */
-	__be32 src_addr;
+	int version;
```

Comment:
According to http://stackoverflow.com/questions/9479424/how-to-determine-what-type-is-used-in-a-union and other similar posts, it's not possible to determine what type is actually used in a union.

We could use a fixed size instead, like in6 and do IPv4 in IPv6 mapping, but that would worsen the case for IPv4 as it consumes four times more memory than necessary.

Another approach might be to pass an extra argument along to all the functions saying whether it's an IPv4 or IPv6 match object being passed.

Third, we could shrink version to a char, saving one byte (I'm presuming int is normally 16 bits).